packages/camera/camera_windows/windows/camera.cpp

```diff
@@ -0,0 +1,271 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "camera.h"
+
+namespace camera_windows {
+using flutter::EncodableList;
+using flutter::EncodableMap;
+using flutter::EncodableValue;
+
+namespace {
+// Camera channel events
+const char kCameraMethodChannelBaseName[] = "flutter.io/cameraPlugin/camera";
+const char kVideoRecordedEvent[] = "video_recorded";
+const char kErrorEvent[] = "error";
+
+// Helper function for creating messaging channel for camera
+std::unique_ptr<flutter::MethodChannel<>> BuildChannelForCamera(
+    flutter::BinaryMessenger *messenger, int64_t camera_id) {
+  auto channel_name =
+      std::string(kCameraMethodChannelBaseName) + std::to_string(camera_id);
+  return std::make_unique<flutter::MethodChannel<>>(
+      messenger, channel_name, &flutter::StandardMethodCodec::GetInstance());
+}
+
+}  // namespace
+
+CameraImpl::CameraImpl(const std::string &device_id)
+    : device_id_(device_id),
+      capture_controller_(nullptr),
+      messenger_(nullptr),
+      camera_id_(-1),
+      Camera(device_id) {}
+CameraImpl::~CameraImpl() {
+  capture_controller_ = nullptr;
+
+  SendErrorForPendingResults("Plugin disposed",
+                             "Plugin disposed before request was handled");
+}
+
+void CameraImpl::InitCamera(flutter::TextureRegistrar *texture_registrar,
+                            flutter::BinaryMessenger *messenger,
+                            bool enable_audio,
+                            ResolutionPreset resolution_preset) {
+  auto capture_controller_factory =
+      std::make_unique<CaptureControllerFactoryImpl>();
+  InitCamera(std::move(capture_controller_factory), texture_registrar,
+             messenger, enable_audio, resolution_preset);
+}
+
+void CameraImpl::InitCamera(
+    std::unique_ptr<CaptureControllerFactory> capture_controller_factory,
+    flutter::TextureRegistrar *texture_registrar,
+    flutter::BinaryMessenger *messenger, bool enable_audio,
+    ResolutionPreset resolution_preset) {
+  assert(!device_id_.empty());
+  messenger_ = messenger;
+  capture_controller_ =
+      capture_controller_factory->CreateCaptureController(this);
+  capture_controller_->InitCaptureDevice(texture_registrar, device_id_,
+                                         enable_audio, resolution_preset);
+}
+
+// Adds pending result to the pending_results map.
+// If result already exists, call result error handler
```

Comment:
calls

(Please audit declaration comments for descriptive verb form, per style guide)